python/sbp/msg.py

```diff
@@ -146,6 +174,53 @@ def __repr__(self):
     fmt = "<SBP (preamble=0x%X, msg_type=0x%X, sender=%s, length=%d, payload=%s, crc=0x%X)>"
     return fmt % p
 
+  def _from_binary(self, d):
+    offset = 0
+    for t, s in self.__zips__:
+
+      # numpy dtype
+      if t in TYPES_KEYS_NP:
+        a = np.ndarray(1, TYPES_NP[t], d, offset)
+        offset += a.itemsize
+        res = a.item()
+
+      # array
+      elif t.startswith('array'):
+        # get array item and length
+        splits = t.split(':')
+        item = splits[1]
+        a_size = int(splits[2]) if len(splits) > 2 else None
+        res = []
+        is_np_type = item in TYPES_KEYS_NP
+        item = TYPES_NP[item] if is_np_type else self._get_embedded_type(item)
+
+        # iterate array items
+        while (a_size is None or a_size > 0) and offset < len(d):
+          if is_np_type:
+            a = np.ndarray(1, item, d, offset)
+            offset += a.itemsize
+            res.append(a.item())
+          else:
+            o = item()
+            offset += o.from_binary(d, offset)
+            res.append(o)
+          if a_size is not None:
+            a_size -= 1
+ 
+      # string
+      elif t.startswith('str'):
+        count = int(t.split(':')[1]) if ':' in t else -1
+        res = ''.join(chr(c) for c in np.frombuffer(d, 'u1', count, offset))
+        offset += len(res)
+        res = res.ljust(count, '\x00')  
+  
+      # custom embedded type       
+      else:
+        res = self._get_embedded_type(t)()
+        offset += res.from_binary(d, offset)
+
+      setattr(self, s, res)
```

Comment:
Can we use more informative variable names on the SBP class?  what is "o",  what is "s" ?